snf-ui/app/controllers/object.js

```diff
@@ -177,14 +184,46 @@ export default Ember.Controller.extend(NameMixin, {
   freezeRenameObject: true,
 
   actions: {
+    saveFile: function(file, content) {
+      file.update(content);
+    },
+
     initAction: function(action){
       this.get('parentController').send('clearSelected');
       this.set('isSelected', true);
       this.send(action);
     },
 
     openEditor: function(){
-      this.send('showDialog', 'editor', this, this.get('model'));
+      var size = this.get('model.size');
```

Comment:
`openEditor` and `openPreviewer` methods are almost identical. consider using a reusable helper method.